x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/XPackSettings.java

```diff
@@ -124,6 +124,9 @@ private XPackSettings() {
     public static final Setting<Boolean> FIPS_MODE_ENABLED =
         Setting.boolSetting("xpack.security.fips_mode.enabled", false, Property.NodeScope);
 
+    /** Setting for enabling or disabling eql. Defaults to true. */
+    public static final Setting<Boolean> EQL_ENABLED = Setting.boolSetting("xpack.eql.enabled", true, Setting.Property.NodeScope);
```

Comment:
This setting already exists in EqlPlugin. Strange that it didn't fail in testing.

Reply:
@imotov should this setting only exist in EqlPlugin for the moment, and when EQL it's ready to go, move the setting to XPackSettings?

Reply:
I think the placement of this setting is not as important as having it in one place only and disabling completely in non-snapshot builds. I don't see any issues with moving it in XPackSettings as long as we disable it there for non-snapshot builds.